mentat/terminal/loading.py

```diff
@@ -0,0 +1,37 @@
+from tqdm import tqdm
+
+from mentat.session_stream import StreamMessage
+
+
+class LoadingHandler:
+    def __init__(self):
+        self.pbar = None
+
+    def update(self, message: StreamMessage):
+        """Create or update a loading bar with text and progress value (0-100)"""
+        if message.extra and message.extra.get("terminate"):
```

Comment:
This isn't really related to your PR, but StreamMessage.extra shouldn't be able to be None; I'm actually going to just change it now since it's only 2 lines.

Reply:
Ooo pyright didn't like that